Реализация Чистой Архитектуры

AlexOtkidach opened the pull request.

AlexOtkidach committed:
1. Разбил на слои

AlexOtkidach committed:
1.Разнес код по пакетам presentation, data — по смыслу.
2.Организовал связь между слоями посредством интерфейсов.

AlexOtkidach committed:
1.Разнес код по пакетам presentation, data — по смыслу.
2.Организовал связь между слоями посредством интерфейсов +

Nick-review commented on .idea/gradle.xml:
Файлы из папки .idea не должны попадать в ваш PR, это индивидуальные настройки вашей IDE и их надо занести в .gitignore

AlexOtkidach replied:
Понял, занес в gitignore

Nick-review commented on app/src/main/java/data/AudioPlayerRepositoryImpl.kt:
```diff
@@ -0,0 +1,50 @@
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class AudioPlayerRepositoryImpl(private val context: Context) : AudioPlayerRepository {
```
в data слое использовать контекст есть нарушение принципа зависимостей, т.к. в android context относится к presentation

Nick-review commented on app/src/main/java/data/AudioPlayerRepositoryImpl.kt:
```diff
@@ -0,0 +1,50 @@
+package data
+
+import domain.Track
```
для data слоя нужна своя моделька, TrackDto, которая потом будет мапиться в Track domain слоя

Nick-review commented on app/src/main/java/data/AudioPlayerRepository.kt:
```diff
@@ -0,0 +1,8 @@
+
+interface AudioPlayerRepository {
+    suspend fun loadTracks(): List<Track>
+    fun checkInternetConnection(): Boolean
```
Проверить интернет коннект вообще не ответственность репозитория, репозиторий должен работать с данными
Это надо в ui перенести или domain

Nick-review commented on app/src/main/java/data/AudioPlayerRepositoryImpl.kt:
```diff
@@ -0,0 +1,50 @@
+        val networkInfo = connectivityManager.activeNetworkInfo
+        return networkInfo != null && networkInfo.isConnected
+    }
+    class TracksRepositoryImpl: GetTracksUseCase {
```
TracksRepository, который наследуется от Use Case это странно. Как минимум, по названиям, репозиторий имплементируем репозитории импл
+ это нигде не используется, это точно нужно сейчас? Я имею ввиду что вызов идёт в view model, а из неё это не вызывается

AlexOtkidach replied:
Согласен, удалил

Nick-review commented on app/src/main/java/data/SharedPreferencesRepository.java:
```diff
@@ -0,0 +1,4 @@
+package data;
+
+public interface SharedPreferencesRepository {
```
А это зачем? Нигде не используется, ещё и в java

Nick-review commented on app/src/main/java/presentation/AudioPlayerActivity.kt:
```diff
@@ -0,0 +1,137 @@
+    private fun adjustTheme() {
+        val nightMode = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
+        val playlistButtonColor = if (nightMode == Configuration.UI_MODE_NIGHT_YES) {
+            R.color.playlist_favorite_icon_color_dark
```
Это всё в идеале должно быть вынесено в тему и не устанавливаться вручную каждый раз

Nick-review commented on app/src/main/java/presentation/AudioPlayerViewModel.kt:
```diff
@@ -0,0 +1,19 @@
+import kotlinx.coroutines.launch
+
+
+class AudioPlayerViewModel(
```
View Model нигде не используется, она точно нужна?

Nick-review commented on app/src/main/java/domain/AudioPlayerUseCase.kt:
```diff
@@ -0,0 +1,22 @@
+import domain.Track
+import data.AudioPlayerRepository
+
+class AudioPlayerUseCase(private val repository: AudioPlayerRepository) : AudioPlayerInteractor {
```
Это неправильно, в domain слоя не должен приходить репозиторий data слоя, это нарушения правила зависимостей
data слоя должен зависеть от domain, у вас наоборот идёт
Назовите его AudioPlayerInteractorImpl если он наследуется от AudioPlayerInteractor

AlexOtkidach replied:
поправил

AlexOtkidach committed:
Update .gitignore to exclude .idea/gradle.xml

AlexOtkidach committed:
Работа со слоями по замечаниям от ревью

AlexOtkidach committed:
Востановил работу приложения после некоторого сбоя в работе.

AlexOtkidach committed:
1. Разбил на слои уже с нормальными зависимостямя
2. Убрал ошибки при воспроизведении
3. Исправил работу счетчика времени воспроизведения
4. Исправил работу с Retrofit

Nick-review commented on app/src/main/java/com/example/plm2/data/TracksRepositoryImpl.kt:
```diff
@@ -0,0 +1,32 @@
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class TracksRepositoryImpl(private val context: Context) : AudioPlayerRepository {
```
Использование контекст убрано, осталось удалить зависимость, передаваемую в конструкторе

Nick-review commented on app/src/main/java/com/example/plm2/domain/Track.kt:
```diff
@@ -38,12 +41,25 @@ data class Track(
     override fun describeContents(): Int {
         return 0
     }
+    fun toTrack(): Track {
```
Не совсем понятно, для чего эту функция нужна и что она делает
